x509/src/certificate_traits.rs

```diff
@@ -0,0 +1,97 @@
+//! CertificateDocument implementation
+
+#[cfg(feature = "alloc")]
+use crate::certificate_document::CertificateDocument;
+
+use crate::Certificate;
+use der::{Error, Result};
+
+#[cfg(feature = "alloc")]
+use der::Document;
+
+#[cfg(feature = "std")]
+use std::path::Path;
+
+#[cfg(feature = "pem")]
+use {alloc::string::String, der::pem::LineEnding};
+
+/// Parse a public key object from an encoded Certificate document.
+pub trait DecodeCertificate: for<'a> TryFrom<Certificate<'a>, Error = Error> + Sized {
```

Comment:
I realize this is following the convention from similar traits in the private key types crates, but I'm not sure this really confers a similar benefit.

Traits like `DecodePrivateKey` and `EncodePrivateKey` exist because there are many potential downstream types which can load and store key data. I'm not sure I see a similar utility here.

Reply:
i agree and only added that for sake of consistency with other. Will drop.

Reply:
Actually, on second look, this seems as good a place as any to hang PEM support. Have left in place for now.

Reply:
I think you can use inherent methods for that. The trait is only impl'd on one type, right?

Reply:
D'oh. Of course. Done.